examples/js/plugins/COGSource.js

```diff
@@ -0,0 +1,123 @@
+/* global itowns, GeoTIFF, COGParser, THREE  */
+
+/**
+ * @classdesc
+ * An object defining the source of resources to get from a [COG]{@link
+ * https://www.cogeo.org/} file. It
+ * inherits from {@link Source}.
+ *
+ * @extends Source
+ *
+ * @example
+ * // Create the source
+ * const cogSource = new itowns.COGSource({
+ *     url: 'https://cdn.jsdelivr.net/gh/iTowns/iTowns2-sample-data/cog/orvault.tif',
+ * });
+ *
+ * // Create the layer
+ * const colorLayer = new itowns.ColorLayer('COG', {
+ *     source: cogSource,
+ * });
+ *
+ * // Add the layer
+ * view.addLayer(colorLayer);
+ */
+class COGSource extends itowns.Source {
+    /**
+     * @param {Object} source - An object that can contain all properties of a
+     * COGSource and {@link Source}. Only `url` is mandatory.
+     *
+     * @constructor
+     */
+    constructor(source) {
+        super(source);
+
+        if (source.zoom) {
+            this.zoom = source.zoom;
+        } else {
+            this.zoom = { min: 0, max: Infinity };
+        }
+
+        this.url = source.url;
+        this.pool = source.pool || new GeoTIFF.Pool();
+        // We don't use fetcher, we let geotiff.js manage it
+        this.fetcher = () => Promise.resolve({});
+        this.parser = COGParser.parse;
+
+        this.defaultNoData = source.defaultNoData || 255;
+        this.defaultTransparent = source.defaultTransparent || 0;
+        this.opaqueByte = source.opaqueByte || 255;
+        this.opaqueFloat = source.opaqueFloat || 1;
+
+        this.whenReady = GeoTIFF.fromUrl(this.url)
+            .then(async (geotiff) => {
+                this.geotiff = geotiff;
+                this.firstImage = await geotiff.getImage();
+                this.origin = this.firstImage.getOrigin();
+                this.noData = (source.noData !== undefined) ? source.noData : this.firstImage.getGDALNoData();
+                this.dataType = this.selectDataType(this.firstImage.getSampleFormat(), this.firstImage.getBitsPerSample());
```

Comment:
Just leaving a comment but I don't know if each band of a geotiff is of the same datatype.

Reply:
No example for now!